packages/suite/src/components/suite/CoinjoinSessionCountdownTimer.tsx

```diff
@@ -1,23 +1,35 @@
-import React from 'react';
-import { useCountdownTimer, useSelector } from '@suite-hooks';
-import { getSessionDeadlineFormat } from '@wallet-utils/coinjoinUtils';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSelector } from '@suite-hooks';
 import { selectRoundsDurationInHours } from '@wallet-reducers/coinjoinReducer';
-import hoursToMilliseconds from 'date-fns/hoursToMilliseconds';
+import differenceInHours from 'date-fns/differenceInHours';
+import { FormattedNumber } from 'react-intl';
 
 interface CoinjoinSessionCountdownTimerProps {
     sessionDeadline: number;
+    unitDisplay?: 'long' | 'short' | 'narrow';
 }
 
 export const CoinjoinSessionCountdownTimer = ({
     sessionDeadline,
+    unitDisplay = 'narrow',
 }: CoinjoinSessionCountdownTimerProps) => {
+    const deadlineDate = useMemo(() => new Date(sessionDeadline), [sessionDeadline]);
     const roundsDurationInHours = useSelector(selectRoundsDurationInHours);
-
-    const { timeLeftString } = useCountdownTimer(
-        sessionDeadline,
-        getSessionDeadlineFormat(sessionDeadline),
-        hoursToMilliseconds(roundsDurationInHours),
+    const getValue = useCallback(
+        (currentTimestamp = Date.now()) =>
+            Math.max(differenceInHours(deadlineDate, currentTimestamp), roundsDurationInHours),
+        [deadlineDate, roundsDurationInHours],
     );
 
-    return <>{timeLeftString}</>;
+    const [hours, setHours] = useState(getValue);
+
+    useEffect(() => {
+        const interval = setInterval(() => {
+            setHours(getValue(Date.now()));
+        }, 1_000);
+
+        return () => clearInterval(interval);
+    }, [deadlineDate, getValue]);
+
+    return <FormattedNumber value={hours} style="unit" unit="hour" unitDisplay={unitDisplay} />;
```

Comment:
This though made me think the timer could be re-written to allow for a `unitDisplay` setting without a need to provide a custom locale to `formatDuration()`

Reply:
I believe having 2 components with timer logic like that is not ideal :(